src/e3/encoding/vex.py

```diff
@@ -0,0 +1,902 @@
+"""Vulnerabilities Exploitability Exchange specifications.
+
+This specification package is based on
+https://www.cisa.gov/sites/default/files/2023-04/minimum-requirements-for-vex-508c.pdf.
+
+The document
+https://www.cisa.gov/sites/default/files/2023-01/VEX_Use_Cases_Aprill2022.pdf
+has also been used for a better understanding of some VEX implementations.
+"""
+
+from __future__ import annotations
+
+from datetime import datetime, timezone
+from dateutil.parser import parse as date_parse
+from enum import Enum
+from typing import TYPE_CHECKING, TypeVar
+from uuid import uuid4
+
+from e3.json import JsonData
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    DocumentSelf = TypeVar("DocumentSelf", bound="Document")
```

Comment:
I don't think this is needed. No need to put type on cls argument in classmethod

Reply:
Done, removed all the `Self` definitions.